sys/net/crosslayer/ng_pktdump/ng_pktdump.c

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (C) 2015 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     net_pktdump
+ * @{
+ *
+ * @file
+ * @brief       Generic module to dump packages received via netapi to STDOUT
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+
+#include "thread.h"
+#include "msg.h"
+#include "net/ng_pktdump.h"
+#include "net/ng_netbase.h"
+
+void _dump_type(ng_nettype_t type)
+{
+    switch (type) {
+        case NG_NETTYPE_UNDEF:
+            printf("NETTYPE_UNDEF (%i)", type);
+            break;
+
+#ifdef MODULE_NG_SIXLOWPAN
+        case NG_NETTYPE_SIXLOWPAN:
+            printf("NETTYPE_SIXLOPAN (%i)", type);
+            break;
+#endif
+#ifdef MODULE_NG_IPV6
+        case NG_NETTYPE_IPV6:
+            printf("NETTYPE_IPV6 (%i)", type);
+            break;
+#endif
+#ifdef MODULE_NG_ICMPV6
+        case NG_NETTYPE_ICMPV6:
+            printf("NETTYPE_ICMPV6 (%i)", type);
+            break;
+#endif
+#ifdef MODULE_NG_TCP
+        case NG_NETTYPE_TCP:
+            printf("NETTYPE_TCP (%i)", type);
+            break;
+#endif
+#ifdef MODULE_NG_UDP
+        case NG_NETTYPE_UDP:
+            printf("NETTYPE_UDP (%i)", type);
+            break;
+#endif
+#ifdef TEST_SUITES
+        case NG_NETTYPE_TEST:
+            printf("NETTYPE_TEST (%i)", type);
+            break;
+#endif
+        default:
+            printf("NETTYPE_UNKNOWN (%i)", type);
+            break;
+    }
+}
+
+void _dump(ng_pktsnip_t *pkt)
+{
+    int snips = 0;
+    int size = 0;
+
+    while (pkt != NULL) {
+        printf("~~ SNIP %2i - size: %3i byte, type: ", snips, pkt->size);
+        _dump_type(pkt->type);
+        puts("");
```

Comment:
Using the `od` module would be a good idea. How about merging this as is in the simple form and improving it then?